src/main/cpp/VideoState.h

```diff
@@ -101,6 +108,39 @@ enum ShowMode {
 	SHOW_MODE_NB
 };
 
+// Note will have to divide by 1000 to get the float value of the rate 
+enum Rates {
+	X1D32,
+	X1D16,
+	X1D8,
+	X1D4,
+	X1D2,
+	X1,
+	X2,
+	X4,
+	X8,
+	X16,
+	X32,
+};
+static const struct RatesEntry {
+	enum Rates rate;
+	float clock_speed;
+	float pts_speed;
+	char *command;
```

Comment:
Please generate the command string on the fly rather than putting it here

Reply:
As I am already looping (time-consuming) in the speed_map to find the correct next rate, I thought this approach might be a faster than including an extra dependency and building a string.